src/user/user.repository.ts

```diff
@@ -0,0 +1,49 @@
+import { Injectable } from '@nestjs/common';
+import { PrismaService } from '../prisma/prisma.service';
+import { Prisma, User } from '@prisma/client';
+import { UserDto } from './dto/user-response';
+
+@Injectable()
+export class UserRepository {
+  constructor(private prisma: PrismaService) {}
+
+  async getUser(where: Prisma.UserWhereUniqueInput): Promise<UserDto> {
```

Comment:
As discussed, I'd remove `where: Prisma.UserWhereUniqueInput` and split this method into two:
- `getUserById(userId: string)`
- `getUserByEmail(email: string)` (if it's used)

I'd do the same for others.

The reason is to not create a query in service that is calling the repository

Reply:
This should be `userId: number` right?